### DISCUSSION TITLE
Redis Replication/Sentinel cluster

### DISCUSSION THREAD

**Discussion by fortra-cloudops-platform** [OT-CONTAINER-KIT/redis-operator#1035] [General]
Has anyone been able to successfully run a Redis cluster using RedisReplication and RedisSentinel? I can get the initial deployment working just fine, but seems something goes wrong as the pods are restarted. At some point the Sentinels cache the IP of the master node even after they have restarted and gotten new pods, which leaves Sentinel failing trying to use an IP currently not assigned to any pod. In other cases I have seen it also get into a state where it thinks the port is 0.0.0.0
I am using redis-operator: v0.18.0 and redis: v7.2.3

**Comment by drivebyer** [comment]
Do you mean that the sentinel will monitor an outdated master IP of the Redis pod?

**Reply by fortra-cloudops-platform** [reply]
I guess. If I restart the RedisReplication pods several times, then check the RedisSentinel logs it is trying to monitor/connect to the IP that had been assigned the master RedisReplication pod. Currently the only workaround I know is to manually run the command to tell Sentinel what the new IP is.

**Reply by drivebyer** [reply]
That is indeed the issue. It's hard to identify the sentinel cluster as the newest pod of the replication cluster because they are in separate custom resources. I opened an issue at https://github.com/OT-CONTAINER-KIT/redis-operator/issues/1003 to address this.

**Reply by fortra-cloudops-platform** [reply]
In your ticket are you suggesting to add Sentinel as a sidecar to the RedisReplication pods?

**Reply by drivebyer** [reply]
No, they run in different workloads but share the same custom resource definition.

**Reply by aneja17** [reply]
Similar issue is what I have observed on sentinel caching old IP - https://github.com/OT-CONTAINER-KIT/redis-operator/issues/1635

@drivebyer does your fix resolve the issue I highlighted above?
